### Import needed packages

In [209]:
import pandas as pd
import numpy as np
import re

### Load all the necessary data

In [210]:
#Load the country-ISO code dataset
country = pd.read_excel('../original_data/country.xlsx')

#Load airport code data and distance data
airport_code = pd.read_csv("../original_data/airports.csv")
distance = pd.read_csv("../original_data/Distance_of_All_Airports.csv")

# Load the population by country dataset
pop = pd.read_csv("../original_data/Population.csv")

#Load the climate by country dataset
clim = pd.read_csv("../original_data/Climate.csv")

#Load the currencies by country dataset
curr = pd.read_csv("../original_data/Currencies.csv")

#Load the spoken languages by country dataset
lang = pd.read_csv("../original_data/Languages.csv", encoding='Latin-1')

#Load the religion by country dataset
rel = pd.read_csv("../original_data/Religion.csv")

#Load the peace index dataset
peace = pd.read_csv("../original_data/Peace_Index.csv")

#Load Wef Travel Index data
wef = pd.read_excel('../Original_data/WEF_TTDI.xlsx', header=[0, 1], sheet_name='Index Performance')

# Load average restaurant and accomodation cost data
avg_rest_hot_p = pd.read_csv("../Original_data/Avg_rest_hot_prices.csv")

## EDA

#### Country - ISO Code datset

In [211]:
country.head()

Name Iso3                           Continent     SubContinent
0        Aruba  ABW     Latin America and the Caribbean        Caribbean
1      Andorra  ADO                              Europe  Southern Europe
2  Afghanistan  AFG  South, East and South-Eastern Asia    Southern Asia
3       Angola  AGO                              Africa    Middle Africa
4      Albania  ALB                              Europe  Southern Europe

In [212]:
country.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 329 entries, 0 to 328
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Name          329 non-null    object
 1   Iso3          329 non-null    object
 2   Continent     329 non-null    object
 3   SubContinent  329 non-null    object
dtypes: object(4)
memory usage: 10.4+ KB


#### Airport Distance data

In [213]:
airport_code.head()

Name          City  \
0                               Goroka Airport        Goroka   
1                               Madang Airport        Madang   
2                 Mount Hagen Kagamuga Airport   Mount Hagen   
3                               Nadzab Airport        Nadzab   
4  Port Moresby Jacksons International Airport  Port Moresby   

            Country IATA  ICAO  Latitude   Longitude  
0  Papua New Guinea  GKA  AYGA -6.081690  145.391998  
1  Papua New Guinea  MAG  AYMD -5.207080  145.789002  
2  Papua New Guinea  HGU  AYMH -5.826790  144.296005  
3  Papua New Guinea  LAE  AYNZ -6.569803  146.725977  
4  Papua New Guinea  POM  AYPY -9.443380  147.220001

In [214]:
distance.head()

ORIGIN  ORIGIN_AIRPORT_SEQ_ID DEST  DEST_AIRPORT_SEQ_ID  DISTANCE IN MILES
0    01A                1000101  A43              1005601               30.0
1    06A                1000601  A30              1004301               16.0
2    06A                1000601  A43              1005601               18.0
3    06A                1000601  ADQ              1017004               14.0
4    06A                1000601  AOS              1032401               40.0

In [215]:
airport_distance = distance.merge(airport_code, left_on='ORIGIN', right_on='IATA', how='left')

In [216]:
airport_distance = airport_distance.merge(airport_code, left_on='DEST', right_on='IATA', how='left', suffixes=['_Origin', '_Dest'])

In [217]:
airport_distance

ORIGIN  ORIGIN_AIRPORT_SEQ_ID DEST  DEST_AIRPORT_SEQ_ID  \
0         01A                1000101  A43              1005601   
1         06A                1000601  A30              1004301   
2         06A                1000601  A43              1005601   
3         06A                1000601  ADQ              1017004   
4         06A                1000601  AOS              1032401   
...       ...                    ...  ...                  ...   
328146    ZXU                1635301  OPF              1389806   
328147    ZXU                1635301  PHF              1409803   
328148    ZXU                1635301  SCF              1471201   
328149    ZXU                1635301  TEB              1516702   
328150    ZZV                1636105  SAT              1468305   

        DISTANCE IN MILES                   Name_Origin City_Origin  \
0                    30.0                           NaN         NaN   
1                    16.0                           NaN         NaN   
2                    18.0                           NaN         NaN   
3                    14.0                           NaN         NaN   
4                    40.0                           NaN         NaN   
...                   ...                           ...         ...   
328146             1194.0                           NaN         NaN   
328147              411.0                           NaN         NaN   
328148             2267.0                           NaN         NaN   
328149              147.0                           NaN         NaN   
328150             1182.0  Zanesville Municipal Airport  Zanesville   

       Country_Origin IATA_Origin ICAO_Origin  Latitude_Origin  \
0                 NaN         NaN         NaN              NaN   
1                 NaN         NaN         NaN              NaN   
2                 NaN         NaN         NaN              NaN   
3                 NaN         NaN         NaN              NaN   
4                 NaN         NaN         NaN              NaN   
...               ...         ...         ...              ...   
328146            NaN         NaN         NaN              NaN   
328147            NaN         NaN         NaN              NaN   
328148            NaN         NaN         NaN              NaN   
328149            NaN         NaN         NaN              NaN   
328150  United States         ZZV        KZZV        39.944401   

        Longitude_Origin                                        Name_Dest  \
0                    NaN                                              NaN   
1                    NaN                                              NaN   
2                    NaN                                              NaN   
3                    NaN                                   Kodiak Airport   
4                    NaN                                              NaN   
...                  ...                                              ...   
328146               NaN                      Opa-locka Executive Airport   
328147               NaN  Newport News Williamsburg International Airport   
328148               NaN                               Scottsdale Airport   
328149               NaN                                Teterboro Airport   
328150        -81.892097                San Antonio International Airport   

           City_Dest   Country_Dest IATA_Dest ICAO_Dest  Latitude_Dest  \
0                NaN            NaN       NaN       NaN            NaN   
1                NaN            NaN       NaN       NaN            NaN   
2                NaN            NaN       NaN       NaN            NaN   
3             Kodiak  United States       ADQ      PADQ      57.750000   
4                NaN            NaN       NaN       NaN            NaN   
...              ...            ...       ...       ...            ...   
328146         Miami  United States       OPF      KOPF      25.907000   
328147  Newport News  United States       PHF      

In [218]:
airport_distance.isnull().sum()

ORIGIN                      0
ORIGIN_AIRPORT_SEQ_ID       0
DEST                        0
DEST_AIRPORT_SEQ_ID         0
DISTANCE IN MILES           0
Name_Origin              4875
City_Origin              4875
Country_Origin           4875
IATA_Origin              4875
ICAO_Origin              4875
Latitude_Origin          4875
Longitude_Origin         4875
Name_Dest                4991
City_Dest                4991
Country_Dest             4991
IATA_Dest                4991
ICAO_Dest                4991
Latitude_Dest            4991
Longitude_Dest           4991
dtype: int64

In [219]:
airport_distance.shape

(328151, 19)

#### Population Dataset

In [220]:
pop.head()

Country Name Country Code     Indicator Name Indicator Code  \
0                        Aruba          ABW  Population, total    SP.POP.TOTL   
1  Africa Eastern and Southern          AFE  Population, total    SP.POP.TOTL   
2                  Afghanistan          AFG  Population, total    SP.POP.TOTL   
3   Africa Western and Central          AFW  Population, total    SP.POP.TOTL   
4                       Angola          AGO  Population, total    SP.POP.TOTL   

          1960         1961         1962         1963         1964  \
0      54208.0      55434.0      56234.0      56699.0      57029.0   
1  130836765.0  134159786.0  137614644.0  141202036.0  144920186.0   
2    8996967.0    9169406.0    9351442.0    9543200.0    9744772.0   
3   96396419.0   98407221.0  100506960.0  102691339.0  104953470.0   
4    5454938.0    5531451.0    5608499.0    5679409.0    5734995.0   

          1965  ...         2012         2013         2014         2015  \
0      57357.0  ...     102565.0     103165.0     103776.0     104339.0   
1  148769974.0  ...  547482863.0  562601578.0  578075373.0  593871847.0   
2    9956318.0  ...   31161378.0   32269592.0   33370804.0   34413603.0   
3  107289875.0  ...  370243017.0  380437896.0  390882979.0  401586651.0   
4    5770573.0  ...   25107925.0   26015786.0   26941773.0   27884380.0   

          2016         2017         2018         2019         2020  \
0     104865.0     105361.0     105846.0     106310.0     106766.0   
1  609978946.0  626392880.0  643090131.0  660046272.0  677243299.0   
2   35383028.0   36296111.0   37171922.0   38041757.0   38928341.0   
3  412551299.0  423769930.0  435229381.0  446911598.0  458803476.0   
4   28842482.0   29816769.0   30809787.0   31825299.0   32866268.0   

          2021  
0     107195.0  
1  694665117.0  
2   39835428.0  
3  470898870.0  
4   33933611.0  

[5 rows x 66 columns]

In [221]:
pop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 66 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    266 non-null    object 
 1   Country Code    266 non-null    object 
 2   Indicator Name  266 non-null    object 
 3   Indicator Code  266 non-null    object 
 4   1960            264 non-null    float64
 5   1961            264 non-null    float64
 6   1962            264 non-null    float64
 7   1963            264 non-null    float64
 8   1964            264 non-null    float64
 9   1965            264 non-null    float64
 10  1966            264 non-null    float64
 11  1967            264 non-null    float64
 12  1968            264 non-null    float64
 13  1969            264 non-null    float64
 14  1970            264 non-null    float64
 15  1971            264 non-null    float64
 16  1972            264 non-null    float64
 17  1973            264 non-null    flo

In [222]:
pop.isna().sum() 

Country Name      0
Country Code      0
Indicator Name    0
Indicator Code    0
1960              2
                 ..
2017              2
2018              2
2019              2
2020              2
2021              2
Length: 66, dtype: int64

In [223]:
pop[pop[["2021"]].isnull().any(axis=1)]

Country Name Country Code     Indicator Name Indicator Code       1960  \
69          Eritrea          ERI  Population, total    SP.POP.TOTL  1007586.0   
110  Not classified          INX  Population, total    SP.POP.TOTL        NaN   

          1961       1962       1963       1964       1965  ...  2012  2013  \
69   1033320.0  1060489.0  1088859.0  1118152.0  1148188.0  ...   NaN   NaN   
110        NaN        NaN        NaN        NaN        NaN  ...   NaN   NaN   

     2014  2015  2016  2017  2018  2019  2020  2021  
69    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
110   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  

[2 rows x 66 columns]

Given the exploration done in the pop dataset, we can conclude that for the next step in the cleaning and transformations, we wish to keep only the "Country Name", " Country Code", and the most recent year 2021 (which we will rename to pop_2021). Moreover, to make sure that the dataset is homogeneous we will capitalize the first letter of each word in the "Country Name" column. We can also see that there is two null values in 2021, which will be handled.

After scrutinizing, we found out that the 'Country Code' does not follow ISO-CODE, for example Kosovo and Congo. Therefore, we will connect dataset to our country dataset on Country name, not code.

#### Climate dataset

In [224]:
clim.head()

COUNTRY                                        DESCRIPTION
0     Afghanistan     arid to semiarid; cold winters and hot summers
1        Akrotiri  temperate; Mediterranean with hot, dry summers...
2         Albania  mild temperate; cool, cloudy, wet winters; hot...
3         Algeria  arid to semiarid; mild, wet winters with hot, ...
4  American Samoa  tropical marine, moderated by southeast trade ...

In [225]:
clim.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 277 entries, 0 to 276
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   COUNTRY      277 non-null    object
 1   DESCRIPTION  277 non-null    object
dtypes: object(2)
memory usage: 4.5+ KB


In [226]:
clim.isna().sum()

COUNTRY        0
DESCRIPTION    0
dtype: int64

Given the exploration done in the clim dataset, we can conclude that for the next step in the cleaning and transformations, we wish to keep all the columns, but rename them. Moreover, to make sure that the dataset is homogeneous we will capitalize the first letter of each word in the "COUNTRY" and "DESCRIPTION" column. We can also see that there is no null values, hence no handling of missing values is needed in this case. 

#### Currency dataset

In [227]:
curr.head()

Entity        Currency AlphabeticCode  NumericCode MinorUnit  \
0     AFGHANISTAN         Afghani            AFN        971.0         2   
1   ÅLAND ISLANDS            Euro            EUR        978.0         2   
2         ALBANIA             Lek            ALL          8.0         2   
3         ALGERIA  Algerian Dinar            DZD         12.0         2   
4  AMERICAN SAMOA       US Dollar            USD        840.0         2   

  WithdrawalDate  
0            NaN  
1            NaN  
2            NaN  
3            NaN  
4            NaN

In [228]:
curr.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 441 entries, 0 to 440
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Entity          441 non-null    object 
 1   Currency        441 non-null    object 
 2   AlphabeticCode  438 non-null    object 
 3   NumericCode     435 non-null    float64
 4   MinorUnit       276 non-null    object 
 5   WithdrawalDate  162 non-null    object 
dtypes: float64(1), object(5)
memory usage: 20.8+ KB


In [229]:
curr.isna().sum() 

Entity              0
Currency            0
AlphabeticCode      3
NumericCode         6
MinorUnit         165
WithdrawalDate    279
dtype: int64

In [230]:
curr[curr[["AlphabeticCode"]].isnull().any(axis=1)]

Entity               Currency  \
8                                      ANTARCTICA  No universal currency   
183                           PALESTINE, STATE OF  No universal currency   
222  SOUTH GEORGIA AND THE SOUTH SANDWICH ISLANDS  No universal currency   

    AlphabeticCode  NumericCode MinorUnit WithdrawalDate  
8              NaN          NaN       NaN            NaN  
183            NaN          NaN       NaN            NaN  
222            NaN          NaN       NaN            NaN

In [231]:
curr[curr[["WithdrawalDate"]].notnull().any(axis=1)]

Entity               Currency AlphabeticCode  NumericCode  \
279            AFGHANISTAN                Afghani            AFA          4.0   
280          ÅLAND ISLANDS                 Markka            FIM        246.0   
281                ALBANIA                Old Lek            ALK          8.0   
282                ANDORRA        Andorran Peseta            ADP         20.0   
283                ANDORRA         Spanish Peseta            ESP        724.0   
..                     ...                    ...            ...          ...   
436               ZIMBABWE  Zimbabwe Dollar (new)            ZWN        942.0   
437               ZIMBABWE        Zimbabwe Dollar            ZWR        935.0   
438        ZZ01_Gold-Franc             Gold-Franc            XFO          NaN   
439  ZZ02_RINET Funds Code       RINET Funds Code            XRE          NaN   
440         ZZ05_UIC-Franc              UIC-Franc            XFU          NaN   

    MinorUnit WithdrawalDate  
279       NaN        2003-01  
280       NaN        2002-03  
281       NaN        1989-12  
282       NaN        2003-07  
283       NaN        2002-03  
..        ...            ...  
436       NaN        2006-09  
437       NaN        2009-06  
438       NaN        2006-10  
439       NaN        1999-11  
440       NaN        2013-11  

[162 rows x 6 columns]

Given the exploration done in the curr dataset, we can conclude that for the next step in the cleaning and transformations, we wish to keep only the "Entity", "Currency", and "AlphabeticCode" columns, but rename them. Moreover, to make sure that the dataset is homogeneous we will capitalize the first letter of each word in the "Entity" and "Currency" column. We can also see that there is three null values in the "AlphabeticCode" column, which will be handled. In addition, there are a number of currencies which are no longer used, we will them subset this dataset to only contain currently accepted currencies. 

#### Language dataset

In [232]:
lang.head()

Country                                   Languages Spoken
0  Afghanistan  Dari Persian, Pashtu (both official), other Tu...
1      Albania     Albanian (Tosk is the official dialect), Greek
2      Algeria         Arabic (official), French, Berber dialects
3      Andorra  Catalán (official), French, Castilian, Portug...
4       Angola  Portuguese (official), Bantu and other African...

In [233]:
lang.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 198 entries, 0 to 197
Data columns (total 2 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Country           198 non-null    object
 1   Languages Spoken  198 non-null    object
dtypes: object(2)
memory usage: 3.2+ KB


In [234]:
lang.isna().sum()

Country             0
Languages Spoken    0
dtype: int64

Given the exploration done in the lang dataset, we can conclude that for the next step in the cleaning and transformations, we wish to keep all the columns, but rename them. Moreover, to make sure that the dataset is homogeneous we will capitalize the first letter of each word in the "Country" and "Language Spoken" column. We can also see that there is no null values, hence no handling of missing values is needed in this case. In addition, we can see that the languages spoken includes more than the official languages, in the cleaning we will strive to only keep the official languages. 

#### Religion dataset

In [235]:
rel.head()

Country or Area  Year   Area         Sex     Religion  \
0         Albania  2011  Total  Both Sexes        Total   
1         Albania  2011  Total  Both Sexes      Atheist   
2         Albania  2011  Total  Both Sexes     Catholic   
3         Albania  2011  Total  Both Sexes  Evangelical   
4         Albania  2011  Total  Both Sexes       Muslim   

                              Record Type             Reliability  \
0  Census - de jure - complete tabulation  Final figure, complete   
1  Census - de jure - complete tabulation  Final figure, complete   
2  Census - de jure - complete tabulation  Final figure, complete   
3  Census - de jure - complete tabulation  Final figure, complete   
4  Census - de jure - complete tabulation  Final figure, complete   

   Source Year      Value Value Footnotes  
0       2013.0  2800138.0             NaN  
1       2013.0    69995.0             NaN  
2       2013.0   280921.0             NaN  
3       2013.0     3797.0             NaN  
4       2013.0  1587608.0             NaN

In [236]:
rel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3416 entries, 0 to 3415
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  3416 non-null   object 
 1   Year             3416 non-null   object 
 2   Area             3333 non-null   object 
 3   Sex              3332 non-null   object 
 4   Religion         3332 non-null   object 
 5   Record Type      3332 non-null   object 
 6   Reliability      3332 non-null   object 
 7   Source Year      3332 non-null   float64
 8   Value            3332 non-null   float64
 9   Value Footnotes  1505 non-null   object 
dtypes: float64(2), object(8)
memory usage: 267.0+ KB


In [237]:
rel.isna().sum() 

Country or Area       0
Year                  0
Area                 83
Sex                  84
Religion             84
Record Type          84
Reliability          84
Source Year          84
Value                84
Value Footnotes    1911
dtype: int64

In [238]:
rel[rel[["Religion"]].isnull().any(axis=1)]

Country or Area                                               Year Area  \
3332   footnoteSeqID                                           Footnote  NaN   
3333               1  Because of rounding, totals are not in all cas...  NaN   
3334               2  These data have been randomly rounded to prote...  NaN   
3335               3  Including population in off-shore, migratory a...  NaN   
3336               4  Category 'No religion' includes Agnoticism, At...  NaN   
...              ...                                                ...  ...   
3411              79                 Data of mini-census taken in 2017.  NaN   
3412              80                 Data refer to resident population.  NaN   
3413              81         Excluding population enumerated in hotels.  NaN   
3414              82  Other includes Orthodox, Bahai, Other Christia...  NaN   
3415              83  Excluding Channel Islands (Guernsey and Jersey...  NaN   

      Sex Religion Record Type Reliability  Source Year  Value Value Footnotes  
3332  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3333  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3334  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3335  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3336  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
...   ...      ...         ...         ...          ...    ...             ...  
3411  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3412  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3413  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3414  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3415  NaN      NaN         NaN         NaN          NaN    NaN             NaN  

[84 rows x 10 columns]

In [239]:
rel[rel[["Area"]].isnull().any(axis=1)]

Country or Area                                               Year Area  \
3332   footnoteSeqID                                           Footnote  NaN   
3333               1  Because of rounding, totals are not in all cas...  NaN   
3334               2  These data have been randomly rounded to prote...  NaN   
3335               3  Including population in off-shore, migratory a...  NaN   
3336               4  Category 'No religion' includes Agnoticism, At...  NaN   
...              ...                                                ...  ...   
3411              79                 Data of mini-census taken in 2017.  NaN   
3412              80                 Data refer to resident population.  NaN   
3413              81         Excluding population enumerated in hotels.  NaN   
3414              82  Other includes Orthodox, Bahai, Other Christia...  NaN   
3415              83  Excluding Channel Islands (Guernsey and Jersey...  NaN   

      Sex Religion Record Type Reliability  Source Year  Value Value Footnotes  
3332  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3333  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3334  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3335  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3336  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
...   ...      ...         ...         ...          ...    ...             ...  
3411  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3412  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3413  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3414  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3415  NaN      NaN         NaN         NaN          NaN    NaN             NaN  

[83 rows x 10 columns]

In [240]:
rel[rel[["Sex"]].isnull().any(axis=1)]

Country or Area                                               Year Area  \
3332   footnoteSeqID                                           Footnote  NaN   
3333               1  Because of rounding, totals are not in all cas...  NaN   
3334               2  These data have been randomly rounded to prote...  NaN   
3335               3  Including population in off-shore, migratory a...  NaN   
3336               4  Category 'No religion' includes Agnoticism, At...  NaN   
...              ...                                                ...  ...   
3411              79                 Data of mini-census taken in 2017.  NaN   
3412              80                 Data refer to resident population.  NaN   
3413              81         Excluding population enumerated in hotels.  NaN   
3414              82  Other includes Orthodox, Bahai, Other Christia...  NaN   
3415              83  Excluding Channel Islands (Guernsey and Jersey...  NaN   

      Sex Religion Record Type Reliability  Source Year  Value Value Footnotes  
3332  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3333  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3334  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3335  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3336  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
...   ...      ...         ...         ...          ...    ...             ...  
3411  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3412  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3413  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3414  NaN      NaN         NaN         NaN          NaN    NaN             NaN  
3415  NaN      NaN         NaN         NaN          NaN    NaN             NaN  

[84 rows x 10 columns]

Given the exploration done in the rel dataset, we can conclude that for the next step in the cleaning and transformations, we wish to keep the "Country or Area" and "Religion" columns but rename them. Moreover, to make sure that the dataset is homogeneous we will capitalize the first letter of each word in the "Country or Area" column. We can also see that there is multiple null values, that arise due to the existing footnotes at the end of the document, which will be dealt with. In addition, we can see that for each country there is the listing of all the religions present in the country. As we only wish to keep the main one the others will be removed. 

#### Peace index dataset

In [241]:
peace.head()

RANK      COUNTRY  SCORE
0   1.0      Iceland  1.100
1   2.0  New Zealand  1.253
2   3.0      Denmark  1.256
3   4.0     Portugal  1.267
4   5.0     Slovenia  1.315

In [242]:
peace.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 177 entries, 0 to 176
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   RANK     163 non-null    float64
 1   COUNTRY  163 non-null    object 
 2   SCORE    163 non-null    float64
dtypes: float64(2), object(1)
memory usage: 4.3+ KB


In [243]:
peace.isna().sum() 

RANK       14
COUNTRY    14
SCORE      14
dtype: int64

In [244]:
peace[peace[["COUNTRY"]].isnull().any(axis=1)]

RANK COUNTRY  SCORE
16    NaN     NaN    NaN
45    NaN     NaN    NaN
74    NaN     NaN    NaN
86    NaN     NaN    NaN
95    NaN     NaN    NaN
108   NaN     NaN    NaN
125   NaN     NaN    NaN
128   NaN     NaN    NaN
138   NaN     NaN    NaN
155   NaN     NaN    NaN
165   NaN     NaN    NaN
168   NaN     NaN    NaN
175   NaN     NaN    NaN
176   NaN     NaN    NaN

Given the exploration done in the peace dataset, we can conclude that for the next step in the cleaning and transformations, we wish to keep the "RANK", and "COUNTRY" columns, but rename them. Moreover, to make sure that the dataset is homogeneous we will capitalize the first letter of each word in the "COUNTRY" column. We can also see that there is multiple null values, that arise due to the existing of completely empty rows, which will be dealt with.

#### WEF Index Data

In [245]:
# to flatten the double header from excel
wef.columns=wef.columns.to_flat_index()

In [246]:
wef.head()

(Unnamed: 0_level_0, ISO Code) (Unnamed: 1_level_0, Economy)  \
0                            JPN                         Japan   
1                            USA                 United States   
2                            ESP                         Spain   
3                            FRA                        France   
4                            DEU                       Germany   

  (Unnamed: 2_level_0, Region) (Unnamed: 3_level_0, Sub Region)  \
0                 Asia-Pacific             Eastern Asia-Pacific   
1                 The Americas        North and Central America   
2           Europe and Eurasia                  Southern Europe   
3           Europe and Eurasia                   Western Europe   
4           Europe and Eurasia                   Western Europe   

  (Unnamed: 4_level_0, Income Group)  \
0              High-income economies   
1              High-income economies   
2              High-income economies   
3              High-income economies   
4              High-income economies   

   (Travel & Tourism Development Index , 2019 Value)  \
0                                           5.209308   
1                                           5.253746   
2                                           5.156125   
3                                           5.139553   
4                                           5.158769   

   (Travel & Tourism Development Index , 2019 Rank)  \
0                                                 2   
1                                                 1   
2                                                 5   
3                                                 6   
4                                                 4   

   (Travel & Tourism Development Index , 2021 Value)  \
0                                           5.245868   
1                                           5.199991   
2                                           5.154733   
3                                           5.126821   
4                                           5.064452   

   (Travel & Tourism Development Index , 2021 Rank)  \
0                                                 1   
1                                                 2   
2                                                 3   
3                                                 4   
4                                                 5   

   (Travel & Tourism Development Index , % Dif Score)  ...  \
0                                           0.007018   ...   
1                                          -0.010232   ...   
2                                          -0.000270   ...   
3                                          -0.002477   ...   
4                                          -0.018283   ...   

   (Socioeconomic Resilience & Conditions pillar, 2021 Value)  \
0                                           5.710859            
1                                           4.496351            
2                                           5.297103            
3                                           5.734719            
4                                           5.884420            

   (Socioeconomic Resilience & Conditions pillar, 2021 Rank)  \
0                                                 11           
1                                                 44           
2                                                 25           
3                                                 10           
4                                                  8           

   (Socioeconomic Resilience & Conditions pillar, % Dif Score)  \
0                                           0.035617             
1                                           0.018056             
2                                           0.068278             
3                                           0.005727             
4                                           0.012092             

   (Socioeconomic Resilience & Conditions pillar, Rank Change)  \
0            

In [247]:
wef.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Columns: 143 entries, ('Unnamed: 0_level_0', 'ISO Code') to ('T&T Demand Pressure & Impact pillar', 'Rank Change')
dtypes: float64(69), int64(69), object(5)
memory usage: 130.8+ KB


In [248]:
wef.isna().sum()

(Unnamed: 0_level_0, ISO Code)                        0
(Unnamed: 1_level_0, Economy)                         0
(Unnamed: 2_level_0, Region)                          0
(Unnamed: 3_level_0, Sub Region)                      0
(Unnamed: 4_level_0, Income Group)                    0
                                                     ..
(T&T Demand Pressure & Impact pillar, 2019 Rank)      0
(T&T Demand Pressure & Impact pillar, 2021 Value)     0
(T&T Demand Pressure & Impact pillar, 2021 Rank)      0
(T&T Demand Pressure & Impact pillar, % Dif Score)    0
(T&T Demand Pressure & Impact pillar, Rank Change)    0
Length: 143, dtype: int64

Among 143 columns we have in dataset, we will choose relevant columns and drop all the others. We found we have no null data in the dataset. Moreover, as the dataset have ISO code itself, there is no further cleaning needed for the country name.

#### Average Restaurant and Accomodation Cost Data

In [249]:
avg_rest_hot_p.head()

Country Name Country Code              Classification Name  \
0         Albania          ALB  Price level index (World = 100)   
1         Algeria          DZA  Price level index (World = 100)   
2  American Samoa          ASM  Price level index (World = 100)   
3          Angola          AGO  Price level index (World = 100)   
4        Anguilla          AIA  Price level index (World = 100)   

  Classification Code                     Series Name  Series Code  \
0               PX.WL  1111000:RESTAURANTS AND HOTELS    1111000.0   
1               PX.WL  1111000:RESTAURANTS AND HOTELS    1111000.0   
2               PX.WL  1111000:RESTAURANTS AND HOTELS    1111000.0   
3               PX.WL  1111000:RESTAURANTS AND HOTELS    1111000.0   
4               PX.WL  1111000:RESTAURANTS AND HOTELS    1111000.0   

  2017 [YR2017]  
0   50.78426859  
1   33.93366644  
2            ..  
3   79.22385228  
4     122.29901

In [250]:
avg_rest_hot_p.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 227 entries, 0 to 226
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Country Name         224 non-null    object 
 1   Country Code         222 non-null    object 
 2   Classification Name  222 non-null    object 
 3   Classification Code  222 non-null    object 
 4   Series Name          222 non-null    object 
 5   Series Code          222 non-null    float64
 6   2017 [YR2017]        222 non-null    object 
dtypes: float64(1), object(6)
memory usage: 12.5+ KB


In [251]:
avg_rest_hot_p.isna().sum() 

Country Name           3
Country Code           5
Classification Name    5
Classification Code    5
Series Name            5
Series Code            5
2017 [YR2017]          5
dtype: int64

In [252]:
avg_rest_hot_p[avg_rest_hot_p[["Country Name"]].isnull().any(axis=1)]

Country Name Country Code Classification Name Classification Code  \
222          NaN          NaN                 NaN                 NaN   
223          NaN          NaN                 NaN                 NaN   
224          NaN          NaN                 NaN                 NaN   

    Series Name  Series Code 2017 [YR2017]  
222         NaN          NaN           NaN  
223         NaN          NaN           NaN  
224         NaN          NaN           NaN

In [253]:
avg_rest_hot_p[avg_rest_hot_p[["Country Code"]].isnull().any(axis=1)]

Country Name Country Code Classification Name  \
222                           NaN          NaN                 NaN   
223                           NaN          NaN                 NaN   
224                           NaN          NaN                 NaN   
225  Data from database: ICP 2017          NaN                 NaN   
226      Last Updated: 10/21/2020          NaN                 NaN   

    Classification Code Series Name  Series Code 2017 [YR2017]  
222                 NaN         NaN          NaN           NaN  
223                 NaN         NaN          NaN           NaN  
224                 NaN         NaN          NaN           NaN  
225                 NaN         NaN          NaN           NaN  
226                 NaN         NaN          NaN           NaN

Given the exploration done in the avg_rest_hot_p dataset, we can conclude that for the next step in the cleaning and transformations, we wish to keep only the "Country Name", " Country Code", and the "2017 [YR2017]" columns, but we will rename them. Moreover, to make sure that the dataset is homogeneous we will capitalize the first letter of each word in the "Country Name" column. We can also see that there is 5 null values, which arise due to empty rows and footnotes, these will be handled accordingly. Additionally, we can see that some countries in the average price have ".." instead of being null, which will be solved.

## Clean, transform, and subset the data

#### Country - ISO CODE Dataset

In [254]:
country['Name'] = country['Name'].str.title()
country['Name'] = country['Name'].str.strip()

#### Airport Distance dataset

In [255]:
airport_distance = airport_distance.dropna(subset=["Name_Origin", "Country_Origin", "Name_Dest", "Country_Dest"], how='any')

In [256]:
airport_distance = airport_distance.drop(columns=['ORIGIN_AIRPORT_SEQ_ID', 'DEST_AIRPORT_SEQ_ID','ICAO_Origin','Latitude_Origin','Longitude_Origin', 'ICAO_Dest', 'Latitude_Dest', 'Longitude_Dest', 'IATA_Origin', 'IATA_Dest'])

In [257]:
airport_distance

ORIGIN DEST  DISTANCE IN MILES                          Name_Origin  \
188       AAO  AUO             2083.0                        Anaco Airport   
189       AAO  BZN             3721.0                        Anaco Airport   
190       AAO  JAX             1826.0                        Anaco Airport   
192       ABE  ABI             1458.0  Lehigh Valley International Airport   
193       ABE  ABQ             1738.0  Lehigh Valley International Airport   
...       ...  ...                ...                                  ...   
328116    ZSA  YWG             2162.0                 San Salvador Airport   
328117    ZSA  YYT             2018.0                 San Salvador Airport   
328118    ZSA  ZRH             4705.0                 San Salvador Airport   
328136    ZXT  UKI               27.0                       Zabrat Airport   
328150    ZZV  SAT             1182.0         Zanesville Municipal Airport   

          City_Origin Country_Origin  \
188             Anaco      Venezuela   
189             Anaco      Venezuela   
190             Anaco      Venezuela   
192         Allentown  United States   
193         Allentown  United States   
...               ...            ...   
328116  Cockburn Town        Bahamas   
328117  Cockburn Town        Bahamas   
328118  Cockburn Town        Bahamas   
328136           Baku     Azerbaijan   
328150     Zanesville  United States   

                                                Name_Dest     City_Dest  \
188                    Auburn University Regional Airport        Auburn   
189                                        Gallatin Field       Bozeman   
190                    Jacksonville International Airport  Jacksonville   
192                              Abilene Regional Airport       Abilene   
193                     Albuquerque International Sunport   Albuquerque   
...                                                   ...           ...   
328116  Winnipeg / James Armstrong Richardson Internat...      Winnipeg   
328117                   St. John's International Airport    St. John's   
328118                                     Zürich Airport        Zurich   
328136                            Ukiah Municipal Airport         Ukiah   
328150                  San Antonio International Airport   San Antonio   

         Country_Dest  
188     United States  
189     United States  
190     United States  
192     United States  
193     United States  
...               ...  
328116         Canada  
328117         Canada  
328118    Switzerland  
328136  United States  
328150  United States  

[319318 rows x 9 columns]

In [258]:
airport_distance.isnull().sum()

ORIGIN               0
DEST                 0
DISTANCE IN MILES    0
Name_Origin          0
City_Origin          0
Country_Origin       0
Name_Dest            0
City_Dest            0
Country_Dest         0
dtype: int64

In [259]:
airport_distance.shape

(319318, 9)

#### Population dataset

In [260]:
# Select the columns we are interest in
pop = pop[["Country Name", "Country Code", "2021"]]

In [261]:
# Rename the columns
pop = pop.rename(columns = {"2021": "Pop_2021", "Country Name": "Country"})

In [262]:
# Capitalise the first letter of each word in the Country column, and remove any extra spaces
pop["Country"] = pop["Country"].str.title()
pop["Country"] = pop["Country"].str.strip()

In [263]:
# Remove empty rows
pop = pop.dropna()

#### Climate dataset

In [264]:
# Rename the columns
clim = clim.rename(columns = {"COUNTRY": "Country", "DESCRIPTION": "Climate"})

In [265]:
# Capitalise the first letter of each word in the Country and Climate column, and remove any extra spaces in the Country column
clim["Country"] = clim["Country"].str.title()
clim["Country"] = clim["Country"].str.strip()
clim["Climate"] = clim["Climate"].str.title()

In [266]:
# Set a new column that classifies climates 
conditions = [(clim["Climate"].str.contains("Tropical|Arid|Semiarid|Desert")),
              (clim["Climate"].str.contains("Temperate|Continental|Hot Summers And Cold Winters" )), 
              (clim["Climate"].str.contains("Mediterranean|Subtropical")),
              (clim["Climate"].str.contains("Highland")), 
              (clim["Climate"].str.contains("Polar|Antarctic")),  
              (clim["Climate"].str.contains("Equatorial"))]

choices = ["Tropical", "Temperate", "Subtropical", "Highland", "Polar", "Equatorial" ]
clim["Climate Zone"] = np.select(conditions, choices, default = "")

#### Currency dataset

In [267]:
# Keep only the rows without a withdrawal date
curr = curr[curr[["WithdrawalDate"]].isnull().any(axis=1)]

In [268]:
# Select the columns to keep
curr = curr[["Entity", "Currency", "AlphabeticCode"]]

In [269]:
# Rename the columns
curr = curr.rename(columns = {"Entity": "Country", "AlphabeticCode": "Currency Code"})

In [270]:
# Capitalise the first letter of each word in the Country and Currency columns, and remove any extra spaces in the country column. Remove any text after "(" in the country column.
curr["Country"] = curr["Country"].str.title()
curr["Country"] = curr['Country'].str.partition("(")[0]
curr["Country"] = curr["Country"].str.strip()
curr["Currency"] = curr["Currency"].str.title()

In [271]:
# Fill the null values with Non Applicable
curr["Currency Code"].fillna("Non Applicable", inplace= True)

#### Language dataset

In [272]:
# Rename the columns
lang = lang.rename(columns = {"Languages Spoken": "Official Language"})

In [273]:
# Capitalise the first letter of each word in the Country and Official Language columns
lang["Country"] = lang["Country"].str.title()
lang["Country"] = lang["Country"].str.strip()
lang["Official Language"] = lang["Official Language"].str.title()

Now we are going to make another language table, that split multiple languages to multiple rows.

In [274]:
def language_cleaning(lan: str):
    # replace 'and' to comma
    # delete numbers and percentage
    try:
        lan = lan.replace(' And ', ',')
        lan = re.sub("[\d%.]", "", lan)

        # first if the text contains 'official', we are going to keep all the official
        if 'Official' in lan:
            lan = lan.partition("(")[0].strip()
            lan = [x.strip() for x in lan.split(',')]

        # otherwise, we are going to keep first one as a main language spoken
        else:
            lan = [lan.partition(",")[0].strip()]

    except:
        print(lan)
        print()
    return lan

In [275]:
# apply language cleaning function to the 'languages Spoken' column
lan_list = lang['Official Language'].apply(language_cleaning)

# split the column that contains a list of multiple languages to multiple columns
# the maximum number of languages spoken was 16, in India
# therefore we make 16 columns
lan_wide = pd.DataFrame(lan_list.to_list(), columns=[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16])


In [276]:
# now we are going to merge lan_wide with original data to have country names in the table
lan_wide = pd.concat([lang, lan_wide], axis=1)

# transform wide format to long format and drop NaN value
lang_long = pd.melt(lan_wide, id_vars='Country', value_vars=[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16]).dropna(subset=['value'])


In [277]:
# change column names
lang_long = lang_long.rename(columns = {"variable": "Number", "value":"Official Language"})

# sort values by Country name
lang_long = lang_long.sort_values(by=['Country', 'Number']).reset_index(drop=True)


In [278]:
lang_long

Country Number Official Language
0                        Afghanistan      1      Dari Persian
1                        Afghanistan      2            Pashtu
2                            Albania      1          Albanian
3                            Algeria      1            Arabic
4                            Andorra      1          Catalã¡N
..                               ...    ...               ...
254                          Vietnam      1        Vietnamese
255  Western Sahara (Proposed State)      1  Hassaniya Arabic
256                            Yemen      1            Arabic
257                           Zambia      1           English
258                         Zimbabwe      1           English

[259 rows x 3 columns]

#### Religion dataset

In [279]:
# Remove the footnote rows, which is where the Religion column 
rel = rel[rel[["Religion"]].notna().any(axis=1)]

In [280]:
# Remove the rows with the total number of religious people. Also remove the other, not stated, and refused to answer religions
rel = rel[rel["Religion"].str.contains("Total|Other|Not Stated|Refused to answer") == False]

In [281]:
# Keep the data from the most recent year for each country
rel = rel.sort_values(by=["Country or Area", "Year"])
rel = rel.drop_duplicates(subset=['Country or Area', "Religion"], keep='last')

In [282]:
# Keep the rows with the highest value, meaning the main religion of each country
rel = rel.sort_values(by=["Country or Area", "Value"])
rel = rel.drop_duplicates(subset=['Country or Area'], keep='last')

In [283]:
# Select the columns to keep
rel = rel[["Country or Area", "Religion"]]

In [284]:
# Rename the columns
rel = rel.rename(columns = {"Country or Area": "Country", "Religion": "Main Religion"})

In [285]:
# Capitalise the first letter of each word in the Country and religion, and remove any extra spaces in the country column.
rel["Country"] = rel["Country"].str.title()
rel["Country"] = rel["Country"].str.strip()
rel["Main Religion"] = rel["Main Religion"].str.title()

#### Peace index dataset

In [286]:
# Select the columns to keep
peace = peace[["RANK", "COUNTRY"]]

In [287]:
# Rename the columns
peace = peace.rename(columns = {"RANK": "Rank", "COUNTRY": "Country"})

In [288]:
# Capitalise the first letter of each word in the Country column
peace["Country"] = peace["Country"].str.title()
peace["Country"] = peace["Country"].str.strip()


In [289]:
# Drop null rows
peace = peace.dropna()

#### WEF Dataset

In [290]:
# Rename columns
wef.rename(columns = {('Unnamed: 0_level_0', 'ISO Code'):'ISO Code',
                      ('Unnamed: 1_level_0', 'Economy'):'Country',
                      ('Unnamed: 2_level_0', 'Region'):'Continent',
                      ('Unnamed: 3_level_0', 'Sub Region'):'Sub Continent',
                      ('Unnamed: 4_level_0', 'Income Group'):'Income Group'
                      }, inplace = True)

In [291]:
# To check the index of the columns that we wish to drop
for i, col in enumerate(wef.columns):
    print(i, col)

0 ISO Code
1 Country
2 Continent
3 Sub Continent
4 Income Group
5 ('Travel & Tourism Development Index ', '2019 Value')
6 ('Travel & Tourism Development Index ', '2019 Rank')
7 ('Travel & Tourism Development Index ', '2021 Value')
8 ('Travel & Tourism Development Index ', '2021 Rank')
9 ('Travel & Tourism Development Index ', '% Dif Score')
10 ('Travel & Tourism Development Index ', 'Rank Change')
11 ('Enabling Environment subindex', '2019 Value')
12 ('Enabling Environment subindex', '2019 Rank')
13 ('Enabling Environment subindex', '2021 Value')
14 ('Enabling Environment subindex', '2021 Rank')
15 ('Enabling Environment subindex', '% Dif Score')
16 ('Enabling Environment subindex', 'Rank Change')
17 ('Travel and Tourism Policy and Enabling\nConditions subindex', '2019 Value')
18 ('Travel and Tourism Policy and Enabling\nConditions subindex', '2019 Rank')
19 ('Travel and Tourism Policy and Enabling\nConditions subindex', '2021 Value')
20 ('Travel and Tourism Policy and Enabling\nCondit

In [292]:
# Drop columns from the behind to prevent index changing
wef2 = wef.iloc[:, [0,1,2,3,4,7,8,25,26,49,50,55,56,79,80,85,86,109,110,115,116,127,128,139,140]]
wef2.columns

Index([                                           'ISO Code',
                                                   'Country',
                                                 'Continent',
                                             'Sub Continent',
                                              'Income Group',
       ('Travel & Tourism Development Index ', '2021 Value'),
        ('Travel & Tourism Development Index ', '2021 Rank'),
                   ('Infrastructure subindex', '2021 Value'),
                    ('Infrastructure subindex', '2021 Rank'),
                ('Safety and Security pillar', '2021 Value'),
                 ('Safety and Security pillar', '2021 Rank'),
                 ('Health and Hygiene pillar', '2021 Value'),
                  ('Health and Hygiene pillar', '2021 Rank'),
             ('International Openness pillar', '2021 Value'),
              ('International Openness pillar', '2021 Rank'),
              ('Price competitiveness pillar', '2021 Value'),
        

#### Average Restaurant and Accomodation Cost Dataset

In [293]:
# Select the columns we are interest in
avg_rest_hot_p = avg_rest_hot_p[["Country Name", "Country Code", "2017 [YR2017]"]]

In [294]:
# Rename the columns
avg_rest_hot_p = avg_rest_hot_p.rename(columns = {"Country Name": "Country", "2017 [YR2017]": "Average Hotel and Restaurant Cost"})

In [295]:
# Capitalise the first letter of each word in the Country column, and remove any extra spaces
avg_rest_hot_p["Country"] = avg_rest_hot_p["Country"].str.title()
avg_rest_hot_p["Country"] = avg_rest_hot_p["Country"].str.strip()

In [296]:
# Remove empty rows
avg_rest_hot_p = avg_rest_hot_p.dropna()

In [297]:
# Substitute ".." for null value
avg_rest_hot_p["Average Hotel and Restaurant Cost"] = avg_rest_hot_p["Average Hotel and Restaurant Cost"].replace("..","")

In [298]:
# Convert the column from string to numeric value
avg_rest_hot_p["Average Hotel and Restaurant Cost"] = pd.to_numeric(avg_rest_hot_p["Average Hotel and Restaurant Cost"], errors = "coerce")

In [299]:
# Round the numbers to two decimal points
avg_rest_hot_p["Average Hotel and Restaurant Cost"] = avg_rest_hot_p["Average Hotel and Restaurant Cost"].round(2)

In [300]:
# Set a new column that classifies countries into budget travel, comfortable travel and luxury travel
conditions = [(avg_rest_hot_p["Average Hotel and Restaurant Cost"] <= 70), ((avg_rest_hot_p["Average Hotel and Restaurant Cost"] > 70) & (avg_rest_hot_p["Average Hotel and Restaurant Cost"] <= 150)), (avg_rest_hot_p["Average Hotel and Restaurant Cost"] > 150)]
choices = ["Budget Traveler", "Comfort traveler", "Luxury Travel"]
avg_rest_hot_p["Type Traveler"] = np.select(conditions, choices, default = "Non Applicable")

## Export cleaned dataset

In [301]:
country.to_csv('../cleaned_data/country.csv')

airport_distance.to_csv('../cleaned_data/airport_distance.csv')

pop.to_csv('../cleaned_data/population.csv')

clim.to_csv('../cleaned_data/climate.csv')

curr.to_csv('../cleaned_data/currencies.csv')

lang.to_csv('../cleaned_data/language.csv')

lang_long.to_csv('../cleaned_data/language_long.csv')

rel.to_csv('../cleaned_data/religion.csv')

peace.to_csv('../cleaned_data/peace_index.csv')

wef2.to_csv('../cleaned_data/wef_ttdi.csv')

avg_rest_hot_p.to_csv("../cleaned_data/avg_rest_hot_p.csv")

## Checking composability
Between country dataset and all the other
based on either 'Country Name' or 'Country Code'

In [302]:
# get cleaned datasets
country = pd.read_csv('../cleaned_data/country.csv', index_col=[0])
climate = pd.read_csv('../cleaned_data/climate.csv', index_col=[0])
currency = pd.read_csv('../cleaned_data/currencies.csv', index_col=[0])
language = pd.read_csv('../cleaned_data/language.csv', encoding='latin-1', index_col=[0])
peace = pd.read_csv('../cleaned_data/peace_index.csv', index_col=[0])
population = pd.read_csv('../cleaned_data/population.csv', index_col=[0])
religion = pd.read_csv('../cleaned_data/Religion.csv', index_col=[0])
wef_ttdi = pd.read_csv('../cleaned_data/wef_ttdi.csv', index_col=[0])
avg_rest_hot_p = pd.read_csv("../cleaned_data/avg_rest_hot_p.csv", index_col=[0])

#### Check climate - country

In [303]:
climate.head()

Country                                            Climate  \
0     Afghanistan     Arid To Semiarid; Cold Winters And Hot Summers   
1        Akrotiri  Temperate; Mediterranean With Hot, Dry Summers...   
2         Albania  Mild Temperate; Cool, Cloudy, Wet Winters; Hot...   
3         Algeria  Arid To Semiarid; Mild, Wet Winters With Hot, ...   
4  American Samoa  Tropical Marine, Moderated By Southeast Trade ...   

  Climate Zone  
0     Tropical  
1    Temperate  
2    Temperate  
3     Tropical  
4     Tropical

In [304]:
climate = climate.merge(country, how='left', left_on='Country', right_on='Name')
climate[climate['Name'].isnull()]

Country  \
1                       Akrotiri   
7                       Anguilla   
8                     Antarctica   
10                  Arctic Ocean   
14   Ashmore And Cartier Islands   
..                           ...   
270               Virgin Islands   
271                  Wake Island   
272            Wallis And Futuna   
273                    West Bank   
274               Western Sahara   

                                               Climate Climate Zone Name Iso3  \
1    Temperate; Mediterranean With Hot, Dry Summers...    Temperate  NaN  NaN   
7         Tropical; Moderated By Northeast Trade Winds     Tropical  NaN  NaN   
8    Severe Low Temperatures Vary With Latitude, El...        Polar  NaN  NaN   
10   Polar Climate Characterized By Persistent Cold...        Polar  NaN  NaN   
14                                            Tropical     Tropical  NaN  NaN   
..                                                 ...          ...  ...  ...   
270  Subtropical, Tempered By Easterly Trade Winds,...  Subtropical  NaN  NaN   
271                                           Tropical     Tropical  NaN  NaN   
272  Tropical; Hot, Rainy Season (November To April...     Tropical  NaN  NaN   
273  Temperate; Temperature And Precipitation Vary ...    Temperate  NaN  NaN   
274  Hot, Dry Desert; Rain Is Rare; Cold Offshore A...     Tropical  NaN  NaN   

    Continent SubContinent  
1         NaN          NaN  
7         NaN          NaN  
8         NaN          NaN  
10        NaN          NaN  
14        NaN          NaN  
..        ...          ...  
270       NaN          NaN  
271       NaN          NaN  
272       NaN          NaN  
273       NaN          NaN  
274       NaN          NaN  

[73 rows x 7 columns]

The rows that cannot be joined with country dataset do not seem like a proper country, so we are going to ignore it.

#### Check Currency - Country

In [305]:
currency = currency.merge(country, how='left', left_on='Country', right_on='Name')
currency[currency['Name'].isnull()]

Country  \
1                                        Åland Islands   
7                                             Anguilla   
8                                           Antarctica   
29                    Bonaire, Sint Eustatius And Saba   
32                                       Bouvet Island   
34                      British Indian Ocean Territory   
49                                    Christmas Island   
50                               Cocos Keeling Islands   
54                The Democratic Republic Of The Congo   
56                                        Cook Islands   
62                                             Curaçao   
76                                            Eswatini   
78                                      European Union   
79                                    Falkland Islands   
86                         French Southern Territories   
92                                           Gibraltar   
96                                          Guadeloupe   
99                                            Guernsey   
105                  Heard Island And Mcdonald Islands   
106                                           Holy See   
113                        International Monetary Fund   
122                                             Jersey   
151                                         Martinique   
154                                            Mayotte   
155  Member Countries Of The African Development Ba...   
158                                         Micronesia   
163                                         Montserrat   
169                                              Nauru   
177                                               Niue   
178                                     Norfolk Island   
184                                Palestine, State Of   
191                                           Pitcairn   
196                                            Réunion   
200                                   Saint Barthélemy   
201       Saint Helena, Ascension And Tristan Da Cunha   
204                                       Saint Martin   
205                          Saint Pierre And Miquelon   
216                                       Sint Maarten   
217  Sistema Unitario De Compensacion Regional De P...   
223       South Georgia And The South Sandwich Islands   
229                             Svalbard And Jan Mayen   
237                       Tanzania, United Republic Of   
241                                            Tokelau   
253               United States Minor Outlying Islands   
265                                     Virgin Islands   
266                                     Virgin Islands   
267                                  Wallis And Futuna   
268                                     Western Sahara   
272              Zz01_Bond Markets Unit European_Eurco   
273              Zz02_Bond Markets Unit European_Emu-6   
274              Zz03_Bond Markets Unit European_Eua-9   
275             Zz04_Bond Markets Unit European_Eua-17   
276                                  Zz06_Testing_Code   
277                                   Zz07_No_Currency   
278                                          Zz08_Gold   
279                                     Zz09_Palladium   
280                                      Zz10_Platinum   
281                                        Zz11_Silver   

                                              Currency   Currency Code Name  \
1                                                 Euro             EUR  NaN   
7                                East Caribbean Dollar             XCD  NaN   
8                                No Universal Currency  Non Applicable  NaN   
29                                           Us Dollar             USD  NaN   
32                                     Norwegian Krone             NOK  NaN   
34                                           Us Dollar             USD  NaN   
49                                   Australian Dollar             AUD 

#### Check language - country

In [306]:
language = language.merge(country, how='left', left_on='Country', right_on='Name')
language[language['Name'].isnull()]

Country  \
41              CãƒÂ´Te D'Ivoire   
50                        East Timor   
116                       Micronesia   
125                            Nauru   
136     Palestinian State (Proposed)   
153     Sã£O Tomã© E Princã­Pe   
193          Vatican City (Holy See)   
196  Western Sahara (Proposed State)   

                                     Official Language Name Iso3 Continent  \
41   French (Official) And African Languages (Dioul...  NaN  NaN       NaN   
50   Tetum, Portuguese (Official); Bahasa Indonesia...  NaN  NaN       NaN   
116  English (Official, Common), Chukese, Pohnpeian...  NaN  NaN       NaN   
125                        Nauruan (Official), English  NaN  NaN       NaN   
136                            Arabic, Hebrew, English  NaN  NaN       NaN   
153                              Portuguese (Official)  NaN  NaN       NaN   
193    Italian, Latin, French, Various Other Languages  NaN  NaN       NaN   
196                  Hassaniya Arabic, Moroccan Arabic  NaN  NaN       NaN   

    SubContinent  
41           NaN  
50           NaN  
116          NaN  
125          NaN  
136          NaN  
153          NaN  
193          NaN  
196          NaN

#### Check peace - country

In [307]:
peace=peace.merge(country, how='left', left_on='Country', right_on='Name')
peace[peace['Name'].isnull()]

Rank                    Country Name Iso3 Continent SubContinent
52    53.0                 The Gambia  NaN  NaN       NaN          NaN
68    69.0                   Eswatini  NaN  NaN       NaN          NaN
138  138.0                  Palestine  NaN  NaN       NaN          NaN
155  155.0  Central African\nRepublic  NaN  NaN       NaN          NaN

#### Check population - country

In [308]:
population=population.merge(country, how='left', left_on='Country', right_on='Name')
population[population['Name'].isnull()]

Country Country Code  \
1                          Africa Eastern And Southern          AFE   
3                           Africa Western And Central          AFW   
7                                           Arab World          ARB   
36                      Central Europe And The Baltics          CEB   
49                              Caribbean Small States          CSS   
61         East Asia & Pacific (Excluding High Income)          EAP   
62                          Early-Demographic Dividend          EAR   
63                                 East Asia & Pacific          EAS   
64       Europe & Central Asia (Excluding High Income)          ECA   
65                               Europe & Central Asia          ECS   
68                                           Euro Area          EMU   
72                                      European Union          EUU   
73            Fragile And Conflict Affected Situations          FCS   
83                                           Gibraltar          GIB   
94                                         High Income          HIC   
97              Heavily Indebted Poor Countries (Hipc)          HPC   
101                                          Ibrd Only          IBD   
102                                   Ida & Ibrd Total          IBT   
103                                          Ida Total          IDA   
104                                          Ida Blend          IDB   
106                                           Ida Only          IDX   
126  Latin America & Caribbean (Excluding High Income)          LAC   
132                          Latin America & Caribbean          LCN   
133       Least Developed Countries: Un Classification          LDC   
134                                         Low Income          LIC   
137                                Lower Middle Income          LMC   
138                                Low & Middle Income          LMY   
140                          Late-Demographic Dividend          LTE   
151                         Middle East & North Africa          MEA   
154                                      Middle Income          MIC   
159  Middle East & North Africa (Excluding High Inc...          MNA   
168                                      North America          NAC   
177                                              Nauru          NRU   
179                                       Oecd Members          OED   
181                                 Other Small States          OSS   
189                           Pre-Demographic Dividend          PRE   
191                          Korea, Dem. People'S Rep.          PRK   
195                        Pacific Island Small States          PSS   
196                          Post-Demographic Dividend          PST   
202                                         South Asia          SAS   
213         Sub-Saharan Africa (Excluding High Income)          SSA   
215                                 Sub-Saharan Africa          SSF   
216                                       Small States          SST   
222                                           Eswatini          SWZ   
228         East Asia & Pacific (Ida & Ibrd Countries)          TEA   
229       Europe & Central Asia (Ida & Ibrd Countries)          TEC   
234  Latin America & The Caribbean (Ida & Ibrd Coun...          TLA   
236  Middle East & North Africa (Ida & Ibrd Countries)          TMN   
238                            South Asia (Ida & Ibrd)          TSA   
239          Sub-Saharan Africa (Ida & Ibrd Countries)          TSS   
247                                Upper Middle Income          UMC   
253                             British Virgin Islands          VGB   
257                                              World          WLD   

         Pop_2021 Name Iso3 Continent SubContinent  
1    6.946651e+08  NaN  NaN       NaN          NaN  
3    4.708989e+08  NaN  NaN       NaN          NaN  
7    4.445178e+08  NaN  NaN       NaN          NaN  


#### Check Religion - country

In [309]:
religion=religion.merge(country, how='left', left_on='Country', right_on='Name')
religion[religion['Name'].isnull()]

Country             Main Religion Name Iso3  \
1                            Anguilla                  Anglican  NaN  NaN   
15             British Virgin Islands                 Methodist  NaN  NaN   
24                       Cook Islands                 Christian  NaN  NaN   
32        Falkland Islands (Malvinas)                 Christian  NaN  NaN   
38                          Gibraltar            Roman Catholic  NaN  NaN   
61   Micronesia (Federated States Of)            Roman Catholic  NaN  NaN   
64                         Montserrat                  Anglican  NaN  NaN   
71                               Niue             Ekalesia Niue  NaN  NaN   
72                     Norfolk Island         Church Of England  NaN  NaN   
86              Saint Helena Ex. Dep.         Church Of England  NaN  NaN   
99                 State Of Palestine                    Islam   NaN  NaN   
104                           Tokelau  Christian Congregational  NaN  NaN   

    Continent SubContinent  
1         NaN          NaN  
15        NaN          NaN  
24        NaN          NaN  
32        NaN          NaN  
38        NaN          NaN  
61        NaN          NaN  
64        NaN          NaN  
71        NaN          NaN  
72        NaN          NaN  
86        NaN          NaN  
99        NaN          NaN  
104       NaN          NaN

#### Check WEF - country

In [310]:
wef_ttdi=wef_ttdi.merge(country, how='left', left_on='ISO Code', right_on='Iso3')
wef_ttdi[wef_ttdi['Name'].isnull()]

Empty DataFrame
Columns: [ISO Code, Country, Continent_x, Sub Continent, Income Group, ('Travel & Tourism Development Index ', '2021 Value'), ('Travel & Tourism Development Index ', '2021 Rank'), ('Infrastructure subindex', '2021 Value'), ('Infrastructure subindex', '2021 Rank'), ('Safety and Security pillar', '2021 Value'), ('Safety and Security pillar', '2021 Rank'), ('Health and Hygiene pillar', '2021 Value'), ('Health and Hygiene pillar', '2021 Rank'), ('International Openness pillar', '2021 Value'), ('International Openness pillar', '2021 Rank'), ('Price competitiveness pillar', '2021 Value'), ('Price competitiveness pillar', '2021 Rank'), ('Natural Resources pillar', '2021 Value'), ('Natural Resources pillar', '2021 Rank'), ('Cultural Resources pillar', '2021 Value'), ('Cultural Resources pillar', '2021 Rank'), ('Environmental Sustainability pillar', '2021 Value'), ('Environmental Sustainability pillar', '2021 Rank'), ('T&T Demand Pressure & Impact pillar', '2021 Value'), ('T&T Demand Pressure & Impact pillar', '2021 Rank'), Name, Iso3, Continent_y, SubContinent]
Index: []

[0 rows x 29 columns]

#### Check Average Restaurant and Accomodation Cost - ISO

In [311]:
avg_rest_hot_p = avg_rest_hot_p.merge(country, how='left', left_on='Country Code', right_on='Iso3')
avg_rest_hot_p [avg_rest_hot_p ['Name'].isnull()]

Country Country Code  \
4                            Anguilla          AIA   
28                            Bonaire          BON   
62                       Cook Islands          COK   
79          East Asia & Pacific (Icp)          EAB   
89        Europe & Central Asia (Icp)          ECB   
151   Latin America & Caribbean (Icp)          LCB   
177  Middle East & North Africa (Icp)          MEB   
184                        Montserrat          MSR   
190                             Nauru          NRU   
200                              Niue          NIU   
201               North America (Icp)          NAB   
247                  South Asia (Icp)          SAB   
257          Sub-Saharan Africa (Icp)          SSB   
281                       Timor-Leste          TLS   
283                           Tokelau          TKL   
313           Virgin Islands, British          VGB   
314                 Wallis And Futuna          WLF   
315                West Bank And Gaza          PSE   
316                             World          WLD   

     Average Hotel and Restaurant Cost     Type Traveler Name Iso3 Continent  \
4                               122.30  Comfort traveler  NaN  NaN       NaN   
28                              156.57     Luxury Travel  NaN  NaN       NaN   
62                                 NaN    Non Applicable  NaN  NaN       NaN   
79                               80.57  Comfort traveler  NaN  NaN       NaN   
89                              120.78  Comfort traveler  NaN  NaN       NaN   
151                              78.35  Comfort traveler  NaN  NaN       NaN   
177                              64.50   Budget Traveler  NaN  NaN       NaN   
184                              84.58  Comfort traveler  NaN  NaN       NaN   
190                                NaN    Non Applicable  NaN  NaN       NaN   
200                                NaN    Non Applicable  NaN  NaN       NaN   
201                             128.94  Comfort traveler  NaN  NaN       NaN   
247                              42.21   Budget Traveler  NaN  NaN       NaN   
257                              46.83   Budget Traveler  NaN  NaN       NaN   
281                                NaN    Non Applicable  NaN  NaN       NaN   
283                                NaN    Non Applicable  NaN  NaN       NaN   
313                             116.88  Comfort traveler  NaN  NaN       NaN   
314                                NaN    Non Applicable  NaN  NaN       NaN   
315                              96.53  Comfort traveler  NaN  NaN       NaN   
316                             100.00  Comfort traveler  NaN  NaN       NaN   

    SubContinent  
4            NaN  
28           NaN  
62           NaN  
79           NaN  
89           NaN  
151          NaN  
177          NaN  
184          NaN  
190          NaN  
200          NaN  
201          NaN  
247          NaN  
257          NaN  
281          NaN  
283          NaN  
313          NaN  
314          NaN  
315          NaN  
316          NaN

## Export cleaned dataset

In [312]:
country.to_csv('../cleaned_data/country.csv')

pop.to_csv('../cleaned_data/population.csv')

clim.to_csv('../cleaned_data/climate.csv')

curr.to_csv('../cleaned_data/currencies.csv')

lang.to_csv('../cleaned_data/language.csv')

rel.to_csv('../cleaned_data/religion.csv')

peace.to_csv('../cleaned_data/peace_index.csv')

wef2.to_csv('../cleaned_data/wef_ttdi.csv')

avg_rest_hot_p.to_csv("../cleaned_data/avg_rest_hot_p.csv")

## Checking composability
Between country dataset and all the other
based on either 'Country Name' or 'Country Code'

In [313]:
# get cleaned datasets
country = pd.read_csv('../cleaned_data/country.csv', index_col=[0])
climate = pd.read_csv('../cleaned_data/climate.csv', index_col=[0])
currency = pd.read_csv('../cleaned_data/currencies.csv', index_col=[0])
language = pd.read_csv('../cleaned_data/language.csv', encoding='latin-1', index_col=[0])
peace = pd.read_csv('../cleaned_data/peace_index.csv', index_col=[0])
population = pd.read_csv('../cleaned_data/population.csv', index_col=[0])
religion = pd.read_csv('../cleaned_data/Religion.csv', index_col=[0])
wef_ttdi = pd.read_csv('../cleaned_data/wef_ttdi.csv', index_col=[0])
avg_rest_hot_p = pd.read_csv("../cleaned_data/avg_rest_hot_p.csv", index_col=[0])

#### Check climate - country

In [314]:
climate.head()

Country                                            Climate  \
0     Afghanistan     Arid To Semiarid; Cold Winters And Hot Summers   
1        Akrotiri  Temperate; Mediterranean With Hot, Dry Summers...   
2         Albania  Mild Temperate; Cool, Cloudy, Wet Winters; Hot...   
3         Algeria  Arid To Semiarid; Mild, Wet Winters With Hot, ...   
4  American Samoa  Tropical Marine, Moderated By Southeast Trade ...   

  Climate Zone  
0     Tropical  
1    Temperate  
2    Temperate  
3     Tropical  
4     Tropical

In [315]:
climate = climate.merge(country, how='left', left_on='Country', right_on='Name')
climate[climate['Name'].isnull()]

Country  \
1                       Akrotiri   
7                       Anguilla   
8                     Antarctica   
10                  Arctic Ocean   
14   Ashmore And Cartier Islands   
..                           ...   
270               Virgin Islands   
271                  Wake Island   
272            Wallis And Futuna   
273                    West Bank   
274               Western Sahara   

                                               Climate Climate Zone Name Iso3  \
1    Temperate; Mediterranean With Hot, Dry Summers...    Temperate  NaN  NaN   
7         Tropical; Moderated By Northeast Trade Winds     Tropical  NaN  NaN   
8    Severe Low Temperatures Vary With Latitude, El...        Polar  NaN  NaN   
10   Polar Climate Characterized By Persistent Cold...        Polar  NaN  NaN   
14                                            Tropical     Tropical  NaN  NaN   
..                                                 ...          ...  ...  ...   
270  Subtropical, Tempered By Easterly Trade Winds,...  Subtropical  NaN  NaN   
271                                           Tropical     Tropical  NaN  NaN   
272  Tropical; Hot, Rainy Season (November To April...     Tropical  NaN  NaN   
273  Temperate; Temperature And Precipitation Vary ...    Temperate  NaN  NaN   
274  Hot, Dry Desert; Rain Is Rare; Cold Offshore A...     Tropical  NaN  NaN   

    Continent SubContinent  
1         NaN          NaN  
7         NaN          NaN  
8         NaN          NaN  
10        NaN          NaN  
14        NaN          NaN  
..        ...          ...  
270       NaN          NaN  
271       NaN          NaN  
272       NaN          NaN  
273       NaN          NaN  
274       NaN          NaN  

[73 rows x 7 columns]

The rows that cannot be joined with country dataset do not seem like a proper country, so we are going to ignore it.

#### Check Currency - Country

In [316]:
currency = currency.merge(country, how='left', left_on='Country', right_on='Name')
currency[currency['Name'].isnull()]

Country  \
1                                        Åland Islands   
7                                             Anguilla   
8                                           Antarctica   
29                    Bonaire, Sint Eustatius And Saba   
32                                       Bouvet Island   
34                      British Indian Ocean Territory   
49                                    Christmas Island   
50                               Cocos Keeling Islands   
54                The Democratic Republic Of The Congo   
56                                        Cook Islands   
62                                             Curaçao   
76                                            Eswatini   
78                                      European Union   
79                                    Falkland Islands   
86                         French Southern Territories   
92                                           Gibraltar   
96                                          Guadeloupe   
99                                            Guernsey   
105                  Heard Island And Mcdonald Islands   
106                                           Holy See   
113                        International Monetary Fund   
122                                             Jersey   
151                                         Martinique   
154                                            Mayotte   
155  Member Countries Of The African Development Ba...   
158                                         Micronesia   
163                                         Montserrat   
169                                              Nauru   
177                                               Niue   
178                                     Norfolk Island   
184                                Palestine, State Of   
191                                           Pitcairn   
196                                            Réunion   
200                                   Saint Barthélemy   
201       Saint Helena, Ascension And Tristan Da Cunha   
204                                       Saint Martin   
205                          Saint Pierre And Miquelon   
216                                       Sint Maarten   
217  Sistema Unitario De Compensacion Regional De P...   
223       South Georgia And The South Sandwich Islands   
229                             Svalbard And Jan Mayen   
237                       Tanzania, United Republic Of   
241                                            Tokelau   
253               United States Minor Outlying Islands   
265                                     Virgin Islands   
266                                     Virgin Islands   
267                                  Wallis And Futuna   
268                                     Western Sahara   
272              Zz01_Bond Markets Unit European_Eurco   
273              Zz02_Bond Markets Unit European_Emu-6   
274              Zz03_Bond Markets Unit European_Eua-9   
275             Zz04_Bond Markets Unit European_Eua-17   
276                                  Zz06_Testing_Code   
277                                   Zz07_No_Currency   
278                                          Zz08_Gold   
279                                     Zz09_Palladium   
280                                      Zz10_Platinum   
281                                        Zz11_Silver   

                                              Currency   Currency Code Name  \
1                                                 Euro             EUR  NaN   
7                                East Caribbean Dollar             XCD  NaN   
8                                No Universal Currency  Non Applicable  NaN   
29                                           Us Dollar             USD  NaN   
32                                     Norwegian Krone             NOK  NaN   
34                                           Us Dollar             USD  NaN   
49                                   Australian Dollar             AUD 

#### Check language - country

In [317]:
language = language.merge(country, how='left', left_on='Country', right_on='Name')
language[language['Name'].isnull()]

Country  \
41              CãƒÂ´Te D'Ivoire   
50                        East Timor   
116                       Micronesia   
125                            Nauru   
136     Palestinian State (Proposed)   
153     Sã£O Tomã© E Princã­Pe   
193          Vatican City (Holy See)   
196  Western Sahara (Proposed State)   

                                     Official Language Name Iso3 Continent  \
41   French (Official) And African Languages (Dioul...  NaN  NaN       NaN   
50   Tetum, Portuguese (Official); Bahasa Indonesia...  NaN  NaN       NaN   
116  English (Official, Common), Chukese, Pohnpeian...  NaN  NaN       NaN   
125                        Nauruan (Official), English  NaN  NaN       NaN   
136                            Arabic, Hebrew, English  NaN  NaN       NaN   
153                              Portuguese (Official)  NaN  NaN       NaN   
193    Italian, Latin, French, Various Other Languages  NaN  NaN       NaN   
196                  Hassaniya Arabic, Moroccan Arabic  NaN  NaN       NaN   

    SubContinent  
41           NaN  
50           NaN  
116          NaN  
125          NaN  
136          NaN  
153          NaN  
193          NaN  
196          NaN

#### Check peace - country

In [318]:
peace=peace.merge(country, how='left', left_on='Country', right_on='Name')
peace[peace['Name'].isnull()]

KeyError: 'Name'

#### Check population - country

In [ ]:
population=population.merge(country, how='left', left_on='Country', right_on='Name')
population[population['Name'].isnull()]

#### Check Religion - country

In [ ]:
religion=religion.merge(country, how='left', left_on='Country', right_on='Name')
religion[religion['Name'].isnull()]

#### Check WEF - country

In [ ]:
wef_ttdi=wef_ttdi.merge(country, how='left', left_on='ISO Code', right_on='Iso3')
wef_ttdi[wef_ttdi['Name'].isnull()]

#### Check Average Restaurant and Accomodation Cost - ISO

In [ ]:
avg_rest_hot_p = avg_rest_hot_p.merge(country, how='left', left_on='Country Code', right_on='Iso3')
avg_rest_hot_p [avg_rest_hot_p ['Name'].isnull()]

## Export cleaned dataset

In [ ]:
country.to_csv('../cleaned_data/country.csv')

pop.to_csv('../cleaned_data/population.csv')

clim.to_csv('../cleaned_data/climate.csv')

curr.to_csv('../cleaned_data/currencies.csv')

lang.to_csv('../cleaned_data/language.csv')

rel.to_csv('../cleaned_data/religion.csv')

peace.to_csv('../cleaned_data/peace_index.csv')

wef2.to_csv('../cleaned_data/wef_ttdi.csv')

avg_rest_hot_p.to_csv("../cleaned_data/avg_rest_hot_p.csv")

## Checking composability
Between country dataset and all the other
based on either 'Country Name' or 'Country Code'

In [ ]:
# get cleaned datasets
country = pd.read_csv('../cleaned_data/country.csv', index_col=[0])
climate = pd.read_csv('../cleaned_data/climate.csv', index_col=[0])
currency = pd.read_csv('../cleaned_data/currencies.csv', index_col=[0])
language = pd.read_csv('../cleaned_data/language.csv', encoding='latin-1', index_col=[0])
peace = pd.read_csv('../cleaned_data/peace_index.csv', index_col=[0])
population = pd.read_csv('../cleaned_data/population.csv', index_col=[0])
religion = pd.read_csv('../cleaned_data/Religion.csv', index_col=[0])
wef_ttdi = pd.read_csv('../cleaned_data/wef_ttdi.csv', index_col=[0])
avg_rest_hot_p = pd.read_csv("../cleaned_data/avg_rest_hot_p.csv", index_col=[0])

#### Check climate - country

In [ ]:
climate.head()

In [ ]:
climate = climate.merge(country, how='left', left_on='Country', right_on='Name')
climate[climate['Name'].isnull()]

The rows that cannot be joined with country dataset do not seem like a proper country, so we are going to ignore it.

#### Check Currency - Country

In [ ]:
currency = currency.merge(country, how='left', left_on='Country', right_on='Name')
currency[currency['Name'].isnull()]

#### Check language - country

In [ ]:
language = language.merge(country, how='left', left_on='Country', right_on='Name')
language[language['Name'].isnull()]

#### Check peace - country

In [ ]:
peace=peace.merge(country, how='left', left_on='Country', right_on='Name')
peace[peace['Name'].isnull()]

#### Check population - country

In [ ]:
population=population.merge(country, how='left', left_on='Country', right_on='Name')
population[population['Name'].isnull()]

#### Check Religion - country

In [ ]:
religion=religion.merge(country, how='left', left_on='Country', right_on='Name')
religion[religion['Name'].isnull()]

#### Check WEF - country

In [ ]:
wef_ttdi=wef_ttdi.merge(country, how='left', left_on='ISO Code', right_on='Iso3')
wef_ttdi[wef_ttdi['Name'].isnull()]

#### Check Average Restaurant and Accomodation Cost - ISO

In [ ]:
avg_rest_hot_p = avg_rest_hot_p.merge(country, how='left', left_on='Country Code', right_on='Iso3')
avg_rest_hot_p [avg_rest_hot_p ['Name'].isnull()]